torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -103,66 +105,52 @@ IndexCompute::IndexCompute(TensorDomain* td, const std::vector<Val*>& indices) {
 }
 
 std::vector<Val*> IndexCompute::get(
-    TensorDomain* td,
+    const TensorDomain* td,
     const std::vector<Val*>& _indices) {
   IndexCompute ic(td, _indices);
   return ic.indices_;
 }
 
 TensorIndex* Index::getGlobalProducerIndex(
-    TensorView* producer,
-    TensorView* consumer,
+    const TensorView* producer,
+    const TensorView* consumer,
     const std::vector<ForLoop*>& loops) {
-  // This replay will ignore reduction dimensions on the producer
-  auto pind =
-      TransformReplay::replayPasC(producer->domain(), consumer->domain(), -1);
+  // Grab indices from the loops
+  std::vector<Val*> indices(loops.size());
+  std::transform(loops.begin(), loops.end(), indices.begin(), [](ForLoop* fl) {
+    return fl->index();
+  });
 
-  TORCH_INTERNAL_ASSERT(
-      loops.size() == consumer->nDims(),
-      "Dimensionality error in code generator while computing tensor indexes.");
+  // What would the consumer indices be if it was global, keeping in mind
+  // reduction axes:
+  std::vector<Val*> c_inds = IndexCompute::get(consumer->domain(), indices);
 
-  std::vector<ForLoop*> loops_adjusted;
-  size_t it_c = 0, it_p = 0;
-  while (it_c < consumer->nDims() && it_p < pind->noReductions().size()) {
-    if (consumer->axis(it_c)->isBroadcast() &&
-        !pind->noReductions()[it_p]->isBroadcast()) {
-      it_c++;
-    } else {
-      loops_adjusted.push_back(loops[it_c]);
-      it_c++;
-      it_p++;
+  // Computed consumer indices should have everything we need for the producer
+  std::vector<Val*> p_inds;
+  auto p_root = TensorDomain::noReductions(producer->getRootDomain());
+  {
```

Comment:
Any reason for a nested block here?

Reply:
It's for my own safety as I often reuse index names and I have ```size_t it_c = 0, it_p = 0;``` surrounding the while loop. I could probably get rid of them, but I don't think it harms much. thoughts?

Reply:
Perfectly fine to me.

I was suggesting similar things as well for `running_producer` below.